File: app/src/main/java/com/chooloo/www/callmanager/ui/dialpad/DialpadBottomDialogFragment.java
package com.chooloo.www.callmanager.ui.dialpad;

import android.os.Bundle;

import com.chooloo.www.callmanager.ui.base.BaseBottomSheetDialogFragment;

public class DialpadBottomDialogFragment extends BaseBottomSheetDialogFragment {

    public static final String ARG_DIALER = "is_dialer";

    private DialpadFragment.OnKeyDownListener mOnKeyDownListener;
    private DialpadFragment mDialpadFragment;

    public static DialpadBottomDialogFragment newInstance(boolean isDialer) {
        Bundle args = new Bundle();
        args.putBoolean(ARG_DIALER, isDialer);
        DialpadBottomDialogFragment fragment = new DialpadBottomDialogFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onSetup() {
        mDialpadFragment = DialpadFragment.newInstance(getArgsSafely().getBoolean(ARG_DIALER));
        putFragment(mDialpadFragment);
        if (mOnKeyDownListener != null) {
            mDialpadFragment.setOnKeyDownListener(mOnKeyDownListener);
        }
    }

    public void setNumber(String number) {
        mDialpadFragment.setNumber(number);
    }

    public void setOnKeyDownListener(DialpadFragment.OnKeyDownListener onKeyDownListener) {
        mOnKeyDownListener = onKeyDownListener;
    }
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/dialpad/DialpadBottomDialogFragment.kt
package com.chooloo.www.callmanager.ui.dialpad

import android.os.Bundle
import com.chooloo.www.callmanager.ui.base.BaseBottomSheetDialogFragment

class DialpadBottomDialogFragment : BaseBottomSheetDialogFragment() {
    private lateinit var _dialpadFragment: DialpadFragment

    companion object {
        const val ARG_DIALER = "is_dialer"

        @JvmStatic
        fun newInstance(isDialer: Boolean): DialpadBottomDialogFragment {
            return DialpadBottomDialogFragment().apply {
                arguments = Bundle().apply {
                    putBoolean(ARG_DIALER, isDialer)
                }
            }
        }
    }

    var number: String
        get() = _dialpadFragment.number
        set(value) {
            _dialpadFragment.number = value
        }

    override fun onSetup() {
        _dialpadFragment = DialpadFragment.newInstance(argsSafely.getBoolean(ARG_DIALER))
        putFragment(_dialpadFragment)
    }
}